salt/elasticsearch/init.sls

```diff
@@ -0,0 +1,34 @@
+elasticsearch-deps:
+    pkg.installed:
+        - name: apt-transport-https
+
+elasticsearch:
+    # add repo to managed repositories
+    pkgrepo.managed:
+        - name: deb https://artifacts.elastic.co/packages/7.x/apt stable main
+        - key_url: salt://elasticsearch/release-key
+        - require:
+            - pkg: elasticsearch-deps
+    # install elastic search package
+    # TODO versioning
+    pkg.installed:
+        - pkgs:
+            - elasticsearch
+        - require:
+            - pkgrepo: elasticsearch
+
+    # start service and watch the config files for restarting the service
+    service.running:
+        - watch:
+          - file: /etc/elasticsearch/elasticsearch.yml
+          - file: /etc/elasticsearch/jvm.options
+
+/etc/elasticsearch/jvm.options:
```

Comment:
The target file name should be set in the `- name: <filename>` not the statement id, as this makes it very hard to customize the location or refer to it in another statement. Ie this should be `elasticsearch-jvm-options` or similar, and the statement below for the config could probably be put under the `elasticsearch` group (but you can only have one `file` statement per group thus you can't have both there).